src/macaron/repo_finder/repo_finder.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module contains the logic for using/calling the different repo finders."""
+
+import logging
+from collections.abc import Iterator
+
+from packageurl import PackageURL
+
+from macaron.config.defaults import defaults
+from macaron.repo_finder.repo_finder_base import BaseRepoFinder
+from macaron.repo_finder.repo_finder_deps_dev import RepoFinderDepsDev
+from macaron.repo_finder.repo_finder_java import JavaRepoFinder
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+def find_repo(purl_string: str) -> Iterator[str]:
+    """Retrieve the repository URL that matches the given PURL.
+
+    Parameters
+    ----------
+    purl_string : str
+        The purl string representing a package.
+
+    Yields
+    ------
+    Iterator[str] :
+        The repository URLs found for the passed package.
+    """
+    # Parse the purl string
+    try:
+        purl = PackageURL.from_string(purl_string)
+    except ValueError as error:
+        logger.debug("Invalid PURL: %s, %s", purl_string, error)
+        return
+
+    # Find matching Repo Finder
+    repo_finder: BaseRepoFinder
+    if purl.type == "maven":
+        repo_finder = JavaRepoFinder()
```

Comment:
Does this mean that for software component that uses Maven as the build tool, we always use the JavaRepoFinder (which looks at pom.xml files, etc.) but not the deps dev method?
This behavior is also applied to the dependencies rathered in the `_resolve_more_dependencies` method within `analyzer.py`, isn't it?

Reply:
This is true. But now I will run some more tests on their relative performance.